docs/DecodeExamples.md

```diff
@@ -0,0 +1,76 @@
+## Decode Bytes Examples
+
+Some of the new output variables from `functions` and `event` of the new `v2.1 Dex` needs to be decoded. Here are some propositons in order to do so: <br/>
+
+### Decode `amountsOut` from the `swap` function
+
+The following functions return the decoded values (x, y) extracted from the encoded `bytes32` amounts. The functions are returning the results in a `tuple`.
+
+```typescript
+function decodeAmounts(amounts: Bytes): [bigint, bigint] {
+  /**
+   * Decodes the amounts bytes input as 2 integers.
+   *
+   * @param amounts - amounts to decode.
+   * @return tuple of BigInts with the values decoded.
+   */
+  // Convert amounts to a BigInt
+  const amountsBigInt = BigInt(`0x${Buffer.from(amounts).toString('hex')}`);
+
+  // Read the first 128 bits of the 256 bits
+  const amountsX = amountsBigInt & ((BigInt(2) ** BigInt(128)) - BigInt(1));
+
+  // Read the last 128 bits of the 256 bits
+  const amountsY = amountsBigInt >> BigInt(128);
+
+  return [amountsX, amountsY];
+}
+```
+
+```python
+def decodeAmounts(amounts: bytes) -> tuple:
+    """
+    Decodes the amounts bytes input as 2 integers.
+
+    :param amounts: amounts to decode.
+    :return: tuple of ints with the values decoded.
+    """
```

Comment:
might be good idea to add `amounts = bytes.fromhex(amounts)` here so it works directly with string copied from snowtrace